File: .github/workflows/test_lint.py
import subprocess
import unittest

class TestLinting(unittest.TestCase):
    def test_ruff(self):
        """Test that ruff linting passes without errors."""
        result = subprocess.run(['ruff', 'check', '.'], capture_output=True, text=True)
        self.assertEqual(result.returncode, 0, msg=f'Ruff linting failed:\n{result.stdout}\n{result.stderr}')

if __name__ == '__main__':
    unittest.main()

File: .github/workflows/test_lint.js
const { exec } = require('child_process');
const assert = require('assert');

describe('Linting', function () {
  it('should pass ESLint without errors', function (done) {
    exec('npx eslint .', (error, stdout, stderr) => {
      assert.strictEqual(error, null, `ESLint failed:\n${stdout}\n${stderr}`);
      done();
    });
  });
});